backend/inspirehep/editor/authorlist_utils.py

```diff
@@ -158,6 +179,38 @@ def parse_authors(text, affiliations):
     return authors, warnings
 
 
+def add_author_to_authors(fullname, authors, author_affs):
+    author = []
+    if fullname:
+        author_affs, ids, emails = check_affiliations_for_ids_and_emails(author_affs)
+
+        author.append(fullname)
+        author.append(author_affs)
+        if ids:
+            author.append(ids)
+        if emails:
+            author.append(emails)
+        authors.append(tuple(author))
+    return authors
+
+
+def check_affiliations_for_ids_and_emails(author_affs):
+    ids = []
+    emails = []
+    data_to_remove = []
+    for word in author_affs:
+        if re_is_orcid.match(word):
+            ids.append(["ORCID", word])
+            data_to_remove.append(word)
+        if re.fullmatch(re_is_email, word):
+            data_to_remove.append(word)
+            emails.append(word)
+
+    author_affs = list(set(author_affs) - set(data_to_remove))
```

Comment:
why is that? maybe instead when you discover in a loop that the word is email/id just remove it from `author_affs` and then you wouldn't need all this setlist operations here

Reply:
The problem is that when I do that it doesn't remove all the values that should be removed and not all values are recognized as emails or ids. What if I just add a list for affiliations.
Like:
```
    ids = []
    emails = []
    affiliations = []
    for word in author_affs:
        if re_is_orcid.match(word):
            ids.append(["ORCID", word])
            author_affs.remove(word)
        elif re.fullmatch(re_is_email, word):
            author_affs.remove(word)
            emails.append(word)
        else:
            affiliations.append(word)

    return affiliations, ids, emails
```

Reply:
much better 👍